### PR TITLE
Fix Issue 19687 - Wrong error for overloaded opDispatch + UFCS with non-existing function

### PR DESCRIPTION
**Pull request by RazvanN7**

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
Please target stable, otherwise LGTM.

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -7799,6 +7799,7 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 Expression e = die.semanticY(sc, 1);
                 if (e && isDotOpDispatch(e))
                 {
+                    exp.e2 = exp.e2.expressionSemantic(sc);
```
Comment (RazvanN7) [inline src/dmd/expressionsem.d]: On this branch e2 is semantically analyzed in resolvePropertiesX (below), but that call is done with gagged errors. That is the only time when semantic gets ran on e2, that is why the error never gets to be printed. In order to make sure that UFCS is tried with correct parameters, e2 needs to have semantic ran on it.

**Review by jacob-carlborg — CHANGES_REQUESTED** [review]
Target stable.

**Comment by RazvanN7** [comment]
@jacob-carlborg Done.

**Review by jacob-carlborg — APPROVED** [review]

File: src/dmd/expressionsem.d (hunk 1 above)
Comment (CyberShadow) [reply src/dmd/expressionsem.d]: Thanks! Shouldn't this explanation be in a comment or something though?

Comment (RazvanN7) [reply src/dmd/expressionsem.d]: Done.

**Pull request merged**